proxy/src/schema.rs

```diff
@@ -53,6 +73,52 @@ impl Query {
         "1.0"
     }
 
+    fn commit(ctx: &Context, id: IdInput, sha1: String) -> FieldResult<Option<Commit>> {
+        let repo = GitRepository::new(&ctx.dummy_repo_path).expect("setting up repo failed");
+        let mut browser = GitBrowser::new(&repo).expect("setting up browser for repo failed");
+        browser
+            .commit(radicle_surf::vcs::git::Sha1::new(&sha1))
+            .expect("setting commit failed");
+
+        let history = browser.get_history();
+        let maybe_commit = history.iter().last();
```

Comment:
If you use `commit` the `History` is just going to be the singleton list of that commit. So in this case you can just do `browser.get_history().0.first()` and be guaranteed to get the commit.

Reply:
It's guaranteed to be a commit as it would already fail on the `browser.commit` call?

Reply:
Yup, exactly :)